packages/compiler-cli/src/ngtsc/annotations/directive/src/shared.ts

```
@@ -684,52 +685,138 @@ function parseInputsArray(
-          throw createValueHasWrongTypeError(
-              decorator.node, transformValue, `Input transform must be a function`);
-        }
+    const optionsNode =
+        decorator.args !== null && decorator.args.length === 1 ? decorator.args[0] : undefined;
+    const options = optionsNode !== undefined ? evaluator.evaluate(optionsNode) : null;
+    const required = options instanceof Map ? options.get('required') === true : false;
 
-        transform = parseInputTransformFunction(
-            clazz, classPropertyName, transformValue, reflector, refEmitter);
-      }
-    } else {
+    // To preserve old behavior: Even though TypeScript types ensure proper options are
+    // passed, we sanity check for unsupported values here again.
+    if (options !== null && typeof options !== 'string' && !(options instanceof Map)) {
       throw createValueHasWrongTypeError(
           decorator.node, options,
           `@${decorator.name} decorator argument must resolve to a string or an object literal`);
     }
 
-    inputs[classPropertyName] = {bindingPropertyName, classPropertyName, required, transform};
-  });
+    let alias: string|null = null;
+    if (typeof options === 'string') {
+      alias = options;
+    } else if (options instanceof Map && typeof options.get('alias') === 'string') {
+      alias = options.get('alias') as string;
+    }
+
+    const publicInputName = alias ?? classPropertyName;
+
+    let transform: InputTransform|null = null;
+    if (options instanceof Map && options.has('transform')) {
+      const transformValue = options.get('transform');
+
+      if (!(transformValue instanceof DynamicValue) && !(transformValue instanceof Reference)) {
+        throw createValueHasWrongTypeError(
+            optionsNode!, transformValue, `Input transform must be a function`);
+      }
+
+      transform = parseInputTransformFunction(
+          clazz, classPropertyName, transformValue, reflector, refEmitter);
+    }
+
+    return {
```

⏩ If `@Input` is used as a decorator together with `input`, I think we should report an error that the decorator is not allowed to specify options.

---

Yes, this will be a follow-up and is planned. Marking as resolved for now.